Source/ASCollectionView.mm

```diff
@@ -622,10 +622,16 @@ - (CGSize)sizeForElement:(ASCollectionElement *)element
     return CGSizeZero;
   }
 
+  // Get the indexPath from the pendingMap, as we're about to query the delegate (and it will reference its current data source).
+  // Often this is a translation compared to the visibleMap where the element came from (UICollectionViewLayout requesting sizeForItem).
+  NSIndexPath *indexPath = [_dataController.pendingMap indexPathForElement:element];
   NSString *supplementaryKind = element.supplementaryElementKind;
-  NSIndexPath *indexPath = [_dataController.visibleMap indexPathForElement:element];
   ASSizeRange sizeRange;
-  if (supplementaryKind == nil) {
+  if (indexPath == nil) {
+    // In this case, the latest data no longer has the element, so we can't ask the delegate for its latest constrained size.
+    // The element is still visible, but will be deleted soon. We can reuse the last-known constrainedSize for the element.
```

Comment:
Sounds good. I agree that if we clarified the terminology here, and used it consistently everywhere (and we could even start putting it in method names, or comments at the start of methods), it would be easier to ensure correctness.

Ideas for "UIKit" space:
- View
- UIKit
- Committed

Ideas for "Latest" space:
- Source
- App
- DataSource
- Latest

I feel like there must be a better, more parallel set of names. One trouble with "current" is that you could imagine that being the current UIKit state, or the current App state.